### PR TITLE
Return `NULL` if a datum has a null value

### LINKED ISSUE #52: Query sample for golang fails in some situation
Very nice examples! Helped me a lot when I was using golang!

But a small issue appeared when I tried to query a DB with more than one type of data ingested.

**Description**: When multiple type of data is ingested like Page 8 of the [Official Guidebook](https://docs.aws.amazon.com/timestream/latest/developerguide/timestream.pdf), the query sample will fail with pointer error.

**Reason**: If data in the DB has more than one data type, `NullValue` will be created as a flag and `ScalarValue` will so be not created. When `data[j]` is called by `processScalarType`, `ScalarValue` will be accessed, which causes the nil pointer error (panic: runtime error: invalid memory address or nil pointer dereference)

**Fix**: Just check if the member is nil before the process, as shown here: https://github.com/uruz-7/amazon-timestream-tools/commit/340ae218371f106bd2ba4a994cd23a8ff156c152

### PR DESCRIPTION
**Pull request by tiratatp**
*Issue #, if available:*
This should fix #52 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: sample_apps/go/query-common.go [hunk 1]
```diff
@@ -26,8 +26,11 @@ func write(f *os.File, s string) {
 	}
 }
 
-func processScalarType(data *timestreamquery.Datum) string {
-	return *data.ScalarValue
+func processScalarType(datum *timestreamquery.Datum) string {
+	if datum.NullValue != nil && *datum.NullValue {
+		return "NULL"
```
Comment (lasartej) [inline sample_apps/go/query-common.go:31]: Can you give me an ELI5 of how this “NULL” is used ? Why isn’t it a type as nil or null and instead we use a string representation of it ?

File: sample_apps/go/query-common.go [hunk 2]
```diff
@@ -26,8 +26,11 @@ func write(f *os.File, s string) {
 	}
 }
 
-func processScalarType(data *timestreamquery.Datum) string {
-	return *data.ScalarValue
+func processScalarType(datum *timestreamquery.Datum) string {
+	if datum.NullValue != nil && *datum.NullValue {
```
Comment (lasartej) [inline sample_apps/go/query-common.go:30]: The * signifies a pointer right ? Why do we need these two checks ?

Comment (tiratatp) [reply sample_apps/go/query-common.go:30]: `*datum.NullValue` is a check that `*datum.NullValue` is true (it could be false as well)

File: sample_apps/go/query-common.go (hunk 1 above)
Comment (tiratatp) [reply sample_apps/go/query-common.go:31]: We are displaying a query result as a table. So we need this to be a string.

Comment (lasartej) [reply sample_apps/go/query-common.go:31]: Gotcha

File: sample_apps/go/query-common.go (hunk 2 above)
Comment (lasartej) [reply sample_apps/go/query-common.go:30]: Gotcha - so it's just a short-circuiting

**Review by lasartej — APPROVED** [review]
